graph-ddl/src/test/scala/org/opencypher/graphddl/GraphDdlTest.scala

```diff
@@ -190,8 +191,132 @@ class GraphDdlTest extends FunSpec with Matchers {
+                 |  (A) FROM a,
+                 |  (A)-[B]->(A) FROM b e
+                 |    START NODES (A) FROM a n JOIN ON e.id = n.id
+                 |    END   NODES (A) FROM a n JOIN ON e.id = n.id
+                 |)
+               """.stripMargin),
+      // mixed order
+      GraphDdl("""SET SCHEMA ds1.db1
+                 |CREATE GRAPH myGraph (
+                 |  (A)-[B]->(A) FROM b e
+                 |    START NODES (A) FROM a n JOIN ON e.id = n.id
+                 |    END   NODES (A) FROM a n JOIN ON e.id = n.id,
+                 |  A (x STRING), B (y STRING),
+                 |  (A) FROM a
+                 |)
+               """.stripMargin),
+      // explicit edge type definition
+      GraphDdl("""SET SCHEMA ds1.db1
+                 |CREATE GRAPH myGraph (
+                 |  A (x STRING), B (y STRING),
+                 |  (A)-[B]->(A),
+                 |  (A) FROM a,
+                 |  (A)-[B]->(A) FROM b e
+                 |    START NODES (A) FROM a n JOIN ON e.id = n.id
+                 |    END   NODES (A) FROM a n JOIN ON e.id = n.id
+                 |)
+               """.stripMargin),
+      // pure type definitions extracted to graph type
+      GraphDdl("""SET SCHEMA ds1.db1
+                 |CREATE GRAPH TYPE myType (
+                 |  A (x STRING), B (y STRING),
+                 |  (A), (A)-[B]->(A)
+                 |)
+                 |CREATE GRAPH myGraph OF myType (
+                 |  (A) FROM a,
+                 |  (A)-[B]->(A) FROM b e
+                 |    START NODES (A) FROM a n JOIN ON e.id = n.id
+                 |    END   NODES (A) FROM a n JOIN ON e.id = n.id
+                 |)
+               """.stripMargin),
+      // shadowing
+      GraphDdl("""SET SCHEMA ds1.db1
+                 |CREATE GRAPH TYPE myType (
+                 |  A (x STRING), B (foo STRING),
+                 |  (A), (A)-[B]->(A)
+                 |)
+                 |CREATE GRAPH myGraph OF myType (
+                 |  B (y STRING),
+                 |  (A) FROM a,
+                 |  (A)-[B]->(A) FROM b e
+                 |    START NODES (A) FROM a n JOIN ON e.id = n.id
+                 |    END   NODES (A) FROM a n JOIN ON e.id = n.id
+                 |)
+               """.stripMargin),
+      // only label types in graph type
+      GraphDdl("""SET SCHEMA ds1.db1
+                 |CREATE GRAPH TYPE myType (
+                 |  A (x STRING), B (foo STRING)
+                 |)
+                 |CREATE GRAPH myGraph OF myType (
+                 |  B (y STRING),
+                 |  (A) FROM a,
+                 |  (A)-[B]->(A) FROM b e
+                 |    START NODES (A) FROM a n JOIN ON e.id = n.id
+                 |    END   NODES (A) FROM a n JOIN ON e.id = n.id
+                 |)
+               """.stripMargin)
+    )
+
+    ddls(1) shouldEqual ddls.head
+    ddls(2) shouldEqual ddls.head
+    ddls(3) shouldEqual ddls.head
+    ddls(4) shouldEqual ddls.head
```

Comment:
Missing case 5.

```suggestion
    ddls(4) shouldEqual ddls.head
    ddls(5) shouldEqual ddls.head
```

or even 

```
    ddls.tail.foreach { ddl =>
      ddl shouldEqual ddls.head
    }
```

(won't bother `suggestion`ing that)

Reply:
The suggested method would not tell you which one fails (the line where it fails).

Reply:
Could get fancy with zipWithIndex but why bother